src/main/java/de/tum/cit/aet/artemis/exam/repository/StudentExamRepository.java

```diff
@@ -215,6 +218,38 @@ SELECT COUNT(se)
             """)
     Optional<StudentExam> findByExamIdAndUserId(@Param("examId") long examId, @Param("userId") long userId);
 
+    Optional<StudentExam> findFirstByExamIdAndUserIdOrderByIdDesc(long examId, long userId);
+
+    @Query("""
+            SELECT se
+            FROM StudentExam se
+            JOIN se.studentParticipations p
+            WHERE se.exam.id = :examId
+                AND p.id = :participationId
+            """)
+    Optional<StudentExam> findByExamIdAndParticipationId(@Param("examId") long examId, @Param("participationId") long participationId);
+
+    /**
+     * Return the StudentExam for the given examId and userId, if possible. For test exams, the latest Student Exam is returned.
+     *
+     * @param examId   id of the exam
+     * @param userId   id of the user
+     * @param testExam boolean indicating if the exam is a test exam
+     * @return the student exam
+     * @throws EntityNotFoundException if no student exams could be found
+     */
+    default StudentExam findOneByExamIdAndUserIdElseThrow(long examId, long userId, boolean testExam) {
+        Optional<StudentExam> studentExam;
+        if (testExam) {
+            studentExam = this.findFirstByExamIdAndUserIdOrderByIdDesc(examId, userId);
+        }
+        else {
+            studentExam = this.findByExamIdAndUserId(examId, userId);
+        }
```

Comment:
```suggestion
        Optional<StudentExam> studentExam = this.findFirstByExamIdAndUserIdOrderByIdDesc(examId, userId);
```
Doesn't this work in both cases?

Reply:
yes, I changed it